modules/packages/Sort.chpl

```diff
@@ -802,6 +868,307 @@ proc selectionSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator)
     compilerError("selectionSort() requires 1-D array");
 }
 
+pragma "no doc"
+proc shellSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator,
+               start=Dom.low, end=Dom.high)
+  where !Dom.stridable
+{
+  chpl_check_comparator(comparator, eltType);
+
+  // Based on Sedgewick's Shell Sort -- see
+  // Analysis of Shellsort and Related Algorithms 1996
+  // and see Marcin Ciura - Best Increments for the Average Case of Shellsort
+  // for the choice of these increments.
+  var start_n = start;
```

Comment:
Why are `start_n` and `end_n` necessary? Couldn't we just use `start` and `end` as is?